llama-index-core/llama_index/core/indices/property_graph/sub_retrievers/custom.py

```diff
@@ -0,0 +1,49 @@
+from abc import abstractmethod
+from typing import Any, List
+
+from llama_index.core.graph_stores.types import PropertyGraphStore
+from llama_index.core.indices.property_graph.sub_retrievers.base import BasePGRetriever
+from llama_index.core.schema import NodeWithScore, QueryBundle, TextNode
+
+
+class CustomPGRetriever(BasePGRetriever):
+    """A retriever meant to be easily subclassed to implement custom retrieval logic.
+
+    The user only has to implement:
+    - `init` to initialize the retriever and assign any necessary attributes.
+    - `custom_retrieve` to implement the custom retrieval logic.
+    - `aretrieve_custom` (optional) to implement asynchronous retrieval logic.
+
+    Args:
+        graph_store (PropertyGraphStore):
+            The graph store to retrieve data from.
+    """
+
+    def __init__(
```

Comment:
i played around with the init function and i came up with 

```
def __init__(
        self,
        graph_store: PropertyGraphStore,
        include_text: bool = True,
        **kwargs: Any,
    ) -> None:
        super().__init__(graph_store=graph_store, include_text=include_text, **kwargs)
        self.init(graph_store=graph_store, include_text=include_text, **kwargs)
```

- put self.init after the superclass init (so graph store is initialized)
- make include_text customizable
- always pass in graph store and include_text into self.init 

thoughts?

Reply:
`include_text` kind of relies on having the graph built by llama-index (so I wasn't sure if I should expose that or not)

If we go this route, I almost wonder if the `def init()` function is even needed then?

Reply:
Just make the user subclass the entire init function?

Reply:
mm taking a step back, i guess the user _could_ just define a custom base retriever right? 

so if there's specifically a `CustomPGRetriever` it would inherently be more opinionated. we just have to be clear about the scope of what the user can customize

Reply:
yea they could do that exactly. What I was doing with `CustomPGRetriever` was giving as easy a thing to subclass as possible (limited types that you need to import, easy return types). Do you think that has any value or nah? I was taking some inspiration from CustomQueryEngine here 😄